CI: golangci-lint fails on main due to unchecked errors in test files

### Describe the bug

The `golangci-lint` check currently fails on the `main` branch due to unchecked errors (`errcheck`) in the tests under the `tool/internal/setup` package. This causes local `make lint` tasks (or any CI actions that run `golangci-lint`) to fail with exit status 1.

The specific lint errors reported are:
```text
tool/internal/setup/cleanup_test.go:31:23: Error return value of `stateManager.Track` is not checked (errcheck)
tool/internal/setup/cleanup_test.go:58:23: Error return value of `stateManager.Track` is not checked (errcheck)
tool/internal/setup/pin_test.go:828:13: Error return value of `os.MkdirAll` is not checked (errcheck)
tool/internal/setup/sync_test.go:124:18: Error return value of `mf.AddModuleStmt` is not checked (errcheck)
tool/internal/setup/sync_test.go:125:14: Error return value of `mf.AddGoStmt` is not checked (errcheck)
```

### Environment

 - OS: All
 - Go Version: 1.22+
 - Version: `main` (latest commit `3418108`)

### To Reproduce

Steps to reproduce the behavior:
1. Clone the repository and ensure you are on the `main` branch.
2. Run `make lint` (or directly execute `golangci-lint run ./...`).
3. See the `errcheck` failures reported above.

### Expected behavior

The `golangci-lint` execution should pass successfully with zero warnings on the `main` branch. 
Unused error values returned by standard setup functions like `os.MkdirAll` and `stateManager.Track` in test files should either be asserted with `require.NoError(t, err)` or explicitly ignored (e.g., `_ = ...`) to maintain a healthy and green CI pipeline.

### Additional context

This breaks the build for anyone running `make all` before pushing code and generates unnecessary CI noise. Fixing this issue will involve wrapping the problematic lines with appropriate error assertions in the test suites.

<sub>**Tip**: [React](https://github.blog/news-insights/product-news/add-reactions-to-pull-requests-issues-and-comments/) with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it. Learn more [here](https://opentelemetry.io/community/end-user/issue-participation/).</sub>

--- a/tool/internal/setup/cleanup_test.go
+++ b/tool/internal/setup/cleanup_test.go
@@ -28,7 +28,7 @@ func TestCleanup(t *testing.T) {
 				// track otelc.runtime.go in the state manager so it is removed when Cleanup is called
 				stateManager := NewStateManager()
 				otelcRuntimeGoPath := filepath.Join(dir, OtelcRuntimeFile)
-				stateManager.Track(otelcRuntimeGoPath)
+				require.NoError(t, stateManager.Track(otelcRuntimeGoPath))
 				mustWriteFile(t, otelcRuntimeGoPath, "package main \n\n// dummy runtime file")
 				// The instrumentation package is extracted inside .otelc-build/pkg/,
 				// not at the project root. It is removed as part of .otelc-build/ cleanup.
@@ -55,7 +55,7 @@ func TestCleanup(t *testing.T) {
 				t.Helper()
 				stateManager := NewStateManager()
 				otelcRuntimeGoPath := filepath.Join(dir, OtelcRuntimeFile)
-				stateManager.Track(otelcRuntimeGoPath)
+				require.NoError(t, stateManager.Track(otelcRuntimeGoPath))
 				mustWriteFile(t, otelcRuntimeGoPath, "package main\n\n// dummy runtime file")
 				return ContextWithStateManager(t.Context(), stateManager)
 			},

--- a/tool/internal/setup/pin_test.go
+++ b/tool/internal/setup/pin_test.go
@@ -825,7 +825,7 @@ func TestUpdatePinnedProjects_InvalidRule(t *testing.T) {
 func TestGeneratePinnedProjects(t *testing.T) {
 	dir := t.TempDir()
 	t.Setenv(util.EnvOtelcWorkDir, dir)
-	os.MkdirAll(util.GetBuildTempDir(), 0o755) // ensure .otelc-build exists
+	require.NoError(t, os.MkdirAll(util.GetBuildTempDir(), 0o755)) // ensure .otelc-build exists
 
 	require.NoError(t, os.WriteFile(
 		filepath.Join(dir, "go.mod"),

--- a/tool/internal/setup/sync_test.go
+++ b/tool/internal/setup/sync_test.go
@@ -121,8 +121,8 @@ func TestWriteGoMod(t *testing.T) {
 
 	// Create a modfile
 	mf := &modfile.File{}
-	mf.AddModuleStmt("example.com/test")
-	mf.AddGoStmt("1.21")
+	require.NoError(t, mf.AddModuleStmt("example.com/test"))
+	require.NoError(t, mf.AddGoStmt("1.21"))
 	err := mf.AddRequire("github.com/stretchr/testify", "v1.8.4")
 	require.NoError(t, err)
 

